executor/aggfuncs/func_group_concat.go

```diff
@@ -82,8 +86,17 @@ func (e *groupConcat) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup [
 		}
 	}
 	p.buffer.Truncate(p.buffer.Len() - len(e.sep))
-	// TODO: if total length is greater than global var group_concat_max_len, truncate it.
-	// issue: #7034
+	if e.maxLen > 0 && uint64(p.buffer.Len()) > e.maxLen {
+		i := mathutil.MaxInt
+		if uint64(i) > e.maxLen {
+			i = int(e.maxLen)
+		}
+		p.buffer.Truncate(i)
+		if !e.truncated {
+			sctx.GetSessionVars().StmtCtx.AppendWarning(expression.ErrCutValueGroupConcat)
```

Comment:
Do we need to append the warning for every truncated group?

Reply:
No. The behavior in MySQL is keeps only one : The last row that meets this problem. I'm not sure now how to get the row number for now, so I just keep one warning.